runtime/gc_include/ObjectAllocationAPI.hpp

```diff
@@ -173,6 +173,11 @@ class MM_ObjectAllocationAPI
 						/* Do not zero the TLH if it is already zero's */
 						initializeSlots = initializeSlots && _initializeSlotsOnTLHAllocate;
 #endif /* J9VM_GC_BATCH_CLEAR_TLH */
+#if defined(J9VM_GC_REALTIME)
+						MM_GCRememberedSetFragment *fragment =  &currentThread->sATBBarrierRememberedSetFragment;
+						MM_GCRememberedSet *parent = fragment->fragmentParent;
+						initializeSlots = initializeSlots || (0 != parent->globalFragmentIndex);
+#endif /* J9VM_GC_REALTIME */
```

Comment:
Should code inside be wrapped by run-time check as well? `J9VM_GC_REALTIME` is enabled always

Reply:
Hm, I need to first understand if/why this change is needed.... Why would SATB marking being (in)active affect if allocated object needs to be cleared or not!?

Reply:
ah, I see this comment now that helps:

					/* The metronome pre write barrier might scan this object - always zero it */

Reply:
I think I misread this code. Probably run time check is not required

Reply:
It's probably not required, since the last condition (0 != parent->globalFragmentIndex) is not ever met when SATB barrier is not enabled
But we can think if it is easy to cash ff SATB is active (and keep it updated), in this class instead of fetching it through a few indirect references for each allocation...